web/src/components/pages/contribution/listen/listen.tsx

```diff
@@ -282,26 +288,14 @@ class ListenPage extends React.Component<Props, State> {
           <ContributionPage
             activeIndex={clipIndex}
             demoMode={this.demoMode}
+            hasErrors={!isLoading && (isMissingClips || hasLoadingError)}
             errorContent={
-              !this.props.isLoading &&
-              (clips.length === 0 || !activeClip) && (
-                <div className="empty-container">
-                  <div className="error-card card-dimensions">
-                    <Localized id="listen-empty-state">
-                      <span />
-                    </Localized>
-                    <LinkButton
-                      rounded
-                      to={this.demoMode ? URLS.DEMO_SPEAK : URLS.SPEAK}
-                      className="record-instead">
-                      <MicIcon />{' '}
-                      <Localized id="record-button-label">
-                        <span />
-                      </Localized>
-                    </LinkButton>
-                  </div>
-                </div>
-              )
+              <ListenErrorContent
+                isLoading={isLoading}
+                hasLoadingError={hasLoadingError}
+                isMissingClips={isMissingClips}
+                isDemoMode={this.demoMode}
+              />
```

Comment:
Where is `isLoading` being passed in? It's not one of the props that the `<ListenPage>` component gets in `app.tsx`

Reply:
This comes from where we map the state to the props:

https://github.com/common-voice/common-voice/pull/3483/files/b909af38c3c131b89104b55d77bc37e6104e58eb#diff-b8b5bd64a65dc3163107a35c1a7bc61a03195781b640d8a8dccfc4fc2ec41160R404